lib/tasks/edition.rake

```diff
@@ -0,0 +1,24 @@
+namespace :edition do
+  def delete_edition_files(edition)
+    return unless edition.deleted?
+    edition.attachments.each do |attachment|
+      next if attachment.deleted?
+      puts "deleted edition #{id} has undeleted attachment with url #{attachment.url}"
+    end
+    edition.delete_all_attachments if edition.respond_to?(:delete_all_attachments)
+  end
+
+  desc 'Check if a deleted edition has undeleted attachments, and delete them'
+  task :delete_undeleted_attachments_of_deleted_edition, [:id] => :environment do |_, args|
+    begin
+      delete_edition_files(Edition.find(args[:id]))
+    rescue ActiveRecord::RecordNotFound
+      puts "edition #{id} not found"
+    end
+  end
+
+  desc 'Check if any deleted edition has undeleted attachments, and delete them'
+  task delete_undeleted_attachments_of_all_deleted_editions: :environment do
```

Comment:
I'm not sure, because we want to look at the output of this, which feels more like a task to me.  I expect I'd only want to look at the output of a migration if something went wrong.

Reply:
True, happy to keep it as a task!